tests/components/fritz/test_button.py

```diff
@@ -73,3 +76,127 @@ async def test_buttons(
 
         button = hass.states.get(entity_id)
         assert button.state != STATE_UNKNOWN
+
+
+async def test_wol_button(
+    hass: HomeAssistant,
+    fc_class_mock,
+    fh_class_mock,
+) -> None:
+    """Test Fritz!Tools wake on LAN button."""
+    entry = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
+    entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.helpers.entity.Entity.entity_registry_enabled_default",
+        PropertyMock(return_value=True),
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert entry.state == ConfigEntryState.LOADED
+
+    button = hass.states.get("button.printer_wake_on_lan")
+    assert button
+    assert button.state == STATE_UNKNOWN
+    with patch(
+        "homeassistant.components.fritz.common.AvmWrapper.async_wake_on_lan"
+    ) as mock_press_action:
+        await hass.services.async_call(
+            BUTTON_DOMAIN,
+            SERVICE_PRESS,
+            {ATTR_ENTITY_ID: "button.printer_wake_on_lan"},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        mock_press_action.assert_called_once_with("AA:BB:CC:00:11:22")
+
+        button = hass.states.get("button.printer_wake_on_lan")
+        assert button.state != STATE_UNKNOWN
+
+
+async def test_wol_button_new_device(
+    hass: HomeAssistant,
+    fc_class_mock,
+    fh_class_mock,
+) -> None:
+    """Test WoL button is created for new device at runtime."""
+    entry = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
+    entry.add_to_hass(hass)
+
+    mesh_data = copy.deepcopy(MOCK_MESH_DATA)
+    with (
+        patch(
+            "homeassistant.helpers.entity.Entity.entity_registry_enabled_default",
+            PropertyMock(return_value=True),
+        ),
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+        assert entry.state == ConfigEntryState.LOADED
+
+        assert hass.states.get("button.printer_wake_on_lan")
+
+        mesh_data["nodes"].append(MOCK_NEW_DEVICE_NODE)
+        fh_class_mock.get_mesh_topology.return_value = mesh_data
+
+        async_fire_time_changed(hass, utcnow() + timedelta(seconds=60))
+        await hass.async_block_till_done(wait_background_tasks=True)
+
+        assert hass.states.get("button.server_wake_on_lan")
```

Comment:
It's a bit unclear what the change in state is after the device change. Maybe assert that this state was missing before the device change, if that's what we want to test?